### PR TITLE
Added a example using a generic git repository

### LINKED ISSUE #8: Non GitHub example
The GitHub Terraform provider, comapared to the GitLab or Bitbucket provider, contains a `github_repository_file` resource. This enables GitHub users to easily commit the generated manifest files to the repository that is created by Terraform.

As this is not possible in the other providers, we need an example how to deal with a generic git server. One option may be to just create a null resource that makes the commit when the manifests files change.

### PR DESCRIPTION
**Pull request by nlarzon**
This PR aims to add a example for how to use the provider with a generic git repository.

The example is using a 3rd party provider it is not the best documneted provider but IMO it is better then doing an example using null_resource.

fixes #8

### CODE REVIEW

File: examples/generic/main.tf [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+data "flux_sync" "main" {
+  target_path = "cluster-init"
+  url         = var.repository_url
+}
+
+# Git: Create the files and commit them to the repository
+resource "gitfile_checkout" "example" {
+  repo   = var.repository_url
+  branch = var.branch
+  path   = "${path.root}/git_checkouts/cluster-init"
```
Comment (phillebaba) [inline examples/generic/main.tf:33]: Would this not place the cloned repo inside of your terraform directory?

File: examples/generic/main.tf [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+  target_path = "cluster-init"
+}
+
+data "flux_sync" "main" {
+  target_path = "cluster-init"
+  url         = var.repository_url
+}
+
+# Git: Create the files and commit them to the repository
+resource "gitfile_checkout" "example" {
```
Comment (phillebaba) [inline examples/generic/main.tf:30]: Change resource name to main.

**Comment by phillebaba** [comment]
@nlarzon could you also rebase with main?

File: examples/generic/main.tf (hunk 1 above)
Comment (nlarzon) [reply examples/generic/main.tf:33]: Yes it will, u want it to be moved to path.module ?

File: examples/generic/main.tf (hunk 2 above)
Comment (nlarzon) [reply examples/generic/main.tf]: fixed

File: examples/generic/main.tf (hunk 1 above)
Comment (nlarzon) [reply examples/generic/main.tf]: made it path.module

File: examples/generic/main.tf [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+data "flux_sync" "main" {
+  target_path = "cluster-init"
+  url         = var.repository_url
+}
+
+# Git: Create the files and commit them to the repository
+resource "gitfile_checkout" "main" {
+  repo   = var.repository_url
+  branch = var.branch
+  path   = "${path.module}/git_checkouts/cluster-init"
```
Comment (phillebaba) [inline examples/generic/main.tf:33]: This is still not a good solution as it would put the repo inside of your terraform directory.

Comment (nlarzon) [reply examples/generic/main.tf:33]: yes, I haven't succeeded to use `/tmp` folder and it is hard to determine what happens on a windows machine.

**Review by blaquezzz — APPROVED** [review]

**Comment by eloo** [comment]
Hi, are there any updates for this PR?

**Comment by preston-willis** [comment]
Any updates?

**Pull request closed without merging**